ipfs/mock.go

```diff
@@ -28,3 +33,36 @@ func Mock() APIProvider {
 		return api, nd, nil
 	}
 }
+
+// DagOnlyMock provides an empty APIProvider that only mocks the DAG portion of
+// the ipfs api object. This is much lighter than the full IPFS node and should
+// be favored for CI testing
+func DagOnlyMock() APIProvider {
+	mockAPI := dagOnlyMock{mdutils.Mock()}
+	return func() (coreiface.CoreAPI, io.Closer, error) {
+		return mockAPI, mockAPI, nil
+	}
+}
+
+var _ coreiface.CoreAPI = dagOnlyMock{}
+
+type dagOnlyMock struct {
+	format.DAGService
+}
+
+func (dom dagOnlyMock) Dag() coreiface.APIDagService { return dom }
+
+func (dagOnlyMock) Unixfs() coreiface.UnixfsAPI                                   { return nil }
+func (dagOnlyMock) Block() coreiface.BlockAPI                                     { return nil }
+func (dagOnlyMock) Name() coreiface.NameAPI                                       { return nil }
+func (dagOnlyMock) Key() coreiface.KeyAPI                                         { return nil }
+func (dagOnlyMock) Pin() coreiface.PinAPI                                         { return nil }
+func (dagOnlyMock) Object() coreiface.ObjectAPI                                   { return nil }
+func (dagOnlyMock) Dht() coreiface.DhtAPI                                         { return nil }
+func (dagOnlyMock) Swarm() coreiface.SwarmAPI                                     { return nil }
+func (dagOnlyMock) PubSub() coreiface.PubSubAPI                                   { return nil }
+func (dagOnlyMock) ResolvePath(context.Context, path.Path) (path.Resolved, error) { return nil, nil }
+func (dagOnlyMock) ResolveNode(context.Context, path.Path) (format.Node, error)   { return nil, nil }
+func (dagOnlyMock) WithOptions(...options.ApiOption) (coreiface.CoreAPI, error)   { return nil, nil }
+func (dagOnlyMock) Close() error                                                  { return nil }
+func (dagOnlyMock) Pinning() format.NodeAdder                                     { return nil }
```

Comment:
I'm not the biggest fan of these nil mocks, but it does give us a lot flexibility in the future while also allowing us to not modify `APIProvider`. If we need the full ipfs API for whatever reason, then we can still use the normal `Mock`. If we don't, then we can use the much lighter `dagOnlyMock`

Reply:
tbc, I'm not attached to this solution, and am very open to other ideas for only passing in a dag only mock.

Reply:
It's probably better to actually just pass around the DAG / the absolutely necessary API surface in these cases as well (like @Wondertan did in #352) but I think having different mocks for different purposes might still be useful. Especially as unit-tests most likely will rarely really need the whole CoreAPI object.
I'm OK with this. Would like to hear what @Wondertan thinks about this too.

Reply:
Agree with @liamsi, we can simply deprecate full core API usage. We won't need anything form it except for the DAG.

Reply:
So I don't see any need for `dagOnlyMock`

Reply:
alrighty, the api object has been kicked from the island https://github.com/lazyledger/lazyledger-core/pull/356/commits/1bddff26a78a02675d5118f8310f40cbad831e4e